src/main/java/com/google/flourbot/macro_classes/Macro.java

```diff
@@ -0,0 +1,47 @@
+package com.google.flourbot.macro_classes;
+
+public class Macro {
+    public String creatorId;
+    public String macroName;
+    public String docId;
+    public Trigger macroTrigger;
+    public Action macroAction;
+    
+    public Macro (String creatorId, String macroName, String docId) {
+        this.creatorId = creatorId;
+        this.macroName = macroName;
+        this.docId = docId;
+    }
+
+    public void setCreatorId (String creatorId) {
+        this.creatorId = creatorId;
+    }
+
+    public void setMacroName (String macroName) {
+        this.macroName = macroName;
+    }
+
+    public void setDocId (String docId) {
+        this.docId = docId;
+    }
+
+    public String creatorId () {
+        return creatorId;
+    }
+
+    public String getMacroName () {
+        return macroName;
+    }
+
+    public String getDocId () {
+        return docId;
+    }
+
+    public void createMacroAction (String[] columnValue, String sheetAction, String sheetUrl, String actionType) {
+        macroAction = new Action(columnValue, sheetAction, sheetUrl, actionType);
+    }
+
+    public void createMacroTrigger (String triggerCommand, String triggerType) {
+        macroTrigger = new Trigger (triggerCommand, triggerType);
```

Comment:
Is it ok for the trigger and action properties of a macro to be defined in their own methods? Or should the trigger and action for a macro could be defined in the constructor for the macro? The constructor would need to accept more arguments.

Reply:
Good question! fwiw I had my Macro constructor taking a Trigger object and an Action object.

Reply:
Thanks! I've added them to the constructor.